### ISSUE TITLE
how to monitor connections

### ISSUE THREAD

**Issue by yoormaw** [pivpn/pivpn#77]
hi guys,  firstly - amazing ! this works a treat and was as simple as you said to setup.

my question is, how do I monitor the connections, or attempted connections to my PiVPN ?
is there a log file somewhere, or do I need to setup my router to forward the syslog to the pi, then read the syslogs this way ?    I am keen to block any unwanted 3rd parties trying to break in.

thanks in advance

**Comment by raspicurious** [comment]
Hi

Good to hear you got it working. As to your question @guydeke suggested fail2ban which if I understand correctly might do what you are looking for - http://www.fail2ban.org/wiki/index.php/Main_Page

**Comment by guydeke** [comment]
You can install fail2ban to monitor and ban attackers.
See www.fail2ban.org

command line:
sudo apt-get update
sudo apt-get install fail2ban

With this command you can see the blocked IP's: sudo iptables -L -n
You can see all the previously banned IPs through /var/log/fail2ban.log

I have it on my PIvpn and it works very good.
I configured it that atackers are banned 4ever after a few fails with no time limit between the attacks, so 3 wrong is banned 4ever

**Comment by yoormaw** [comment]
sorry for the delay guys...
many thanks for the info, it sounds exactly what I am looking for and will be giving it a go very soon.
wish me luck with all the coding..... haven't done that since the ZX spectrum days !

**Comment by 0-kaladin** [comment]
You know I used to use fail2ban all the time but never thought of using it for an openvpn server.  For those who use it, one question...  when I used it to block ssh attempts I found basically script kiddies scanning large subnets (that my server was a part of) for port 22 and then their little scripts would run.  Fail2ban would block them and all was well.  Then I started using non standard ssh port and my fail2ban never triggered again.  My point is that unless you are being directly targeted, no one will be scanning all ports for a specific service.  if they want to attach ssh, they will scan 22.  if they want to attack openvpn, they will scan 1194.  
So for those using fail2ban, is your experience the same as mine was for ssh?  if you use a non-standard port is fail2ban never triggering?  I'd expect so.  
So to @yoormaw I'd argue you'd get better security simply not using 1194 as your vpn port.  Then fail2ban is an optional extra layer for the "just in case".
